tests/tests/test_provides_depends.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/python
+# Copyright 2020 Northern.tech AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        https://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import subprocess
+import time
+
+from ..common_setup import enterprise_no_client
+from .common_update import update_image, common_update_procedure
+from .mendertesting import MenderTesting
+from ..MenderAPI import auth, auth_v2, deploy, image, logger
+
+from testutils.infra.device import MenderDevice
+
+
+class TestProvidesDependsEnterprise(MenderTesting):
+    def test_update_provides_depends(self, enterprise_no_client):
+        """
+        Perform two consecutive updates, the first adds virtual provides
+        to the artifact and the second artifact depends on these provides.
+        """
+
+        DEMO_POLL_INTERVAL = 5
+        IMAGE_NAME = "core-image-full-cmdline-qemux86-64.ext4"
+
+        # Create tenant user
+        auth.reset_auth_token()
+        auth.new_tenant("admin", "bob@builder.org", "secret-service")
+        token = auth.current_tenant["tenant_token"]
+
+        # Create client setup with tenant token
+        enterprise_no_client.new_tenant_docker_client("mender-client", token)
+        mender_device = MenderDevice(enterprise_no_client.get_mender_clients()[0])
+        mender_device.host_ip = enterprise_no_client.get_virtual_network_host_ip()
+
+        # Wait for ssh to be open
+        mender_device.ssh_is_opened()
+        # Check that the device has authorized with the backend.
+        device = auth_v2.get_devices(expected_devices=1)
+        device_ids = [device[0]["id"]]
+        auth_v2.accept_devices(1)
+        assert len(auth_v2.get_devices_status("accepted")) == 1
+
+        # Update client with and artifact with custom provides
+        def prepare_provides_artifact(artifact_file, artifact_id):
+            cmd = (
+                # Package MenderAPI in the artifact, just a random folder.
```

Comment:
Please use a different folder, as we have plans in QA backlog to remove this one. Maybe `/testutils` which will survive my refactoring plans or just `/tests/tests`